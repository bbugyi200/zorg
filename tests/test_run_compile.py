"""Tests for the zorg project's 'compile' CLI command."""

from __future__ import annotations

from pathlib import Path

from _pytest.capture import CaptureFixture
from pytest import mark
from syrupy.assertion import SnapshotAssertion as Snapshot
from typist import PathLike

from . import common as c


params = mark.parametrize


@params("zo_path", ["day_log_tmpl"])
def test_compile(
    main: c.MainType,
    capsys: CaptureFixture,
    tmp_path: Path,
    snapshot: Snapshot,
    zo_path: PathLike,
) -> None:
    """Test that the all *.zo test data files compile as expected."""
    zettel_dir = tmp_path / "org"
    zo_with_ext = str(zo_path).replace(".zo", "") + ".zo"
    zo_path = Path(__file__).parent / f"data/{zo_with_ext}"

    argv = [
        "--dir",
        str(zettel_dir),
        "compile",
        str(zo_path),
    ]
    exit_code = main(*argv)

    assert exit_code == 0
    capture = capsys.readouterr()
    assert capture.out == snapshot
