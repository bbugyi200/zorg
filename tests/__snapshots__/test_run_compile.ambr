# serializer version: 1
# name: test_compile[YAMLConfigFile-day_log_tmpl]
  '''
  {'notes': [{'areas': [],
              'body': '%',
              'child_notes': [],
              'contexts': [],
              'create_date': FakeDate(2000, 1, 3),
              'next_note': None,
              'parent_note': None,
              'people': [],
              'prev_note': None,
              'projects': [],
              'properties': {}},
             {'areas': [],
              'body': ' one_day ',
              'child_notes': [],
              'contexts': [],
              'create_date': FakeDate(2000, 1, 3),
              'next_note': None,
              'parent_note': None,
              'people': [],
              'prev_note': None,
              'projects': [],
              'properties': {}},
             {'areas': [],
              'body': '%',
              'child_notes': [],
              'contexts': [],
              'create_date': FakeDate(2000, 1, 3),
              'next_note': None,
              'parent_note': None,
              'people': [],
              'prev_note': None,
              'projects': [],
              'properties': {}}],
   'todos': [{'areas': ['gtd'],
              'body': ' #gtd pomodoro',
              'child_notes': [],
              'contexts': [],
              'create_date': FakeDate(2000, 1, 3),
              'done_date': None,
              'next_note': None,
              'parent_note': None,
              'people': [],
              'prev_note': None,
              'priority': 'C',
              'projects': [],
              'properties': {},
              'status': <TodoStatus.OPEN: 1>}]}
  
  '''
# ---
